Jdp190401 15

pszczepcio opened the pull request.

pszczepcio committed:
- created a relationship with the class Cart and class Order(but I do not have a Order class created)

pszczepcio committed:
- relations between entities were commented on

klikuskodilla commented on src/main/java/com/kodilla/ecommercee/domain/user/User.java:
```diff
@@ -0,0 +1,74 @@
+@Getter
+@Setter
+@Entity
+public class User {
```
Czy ta klasa nie powinna przechowywać też godziny wygenerowanie klucza?

Makedo81 replied:
zeby klucz byl generowany co godzine to powinno byc nad metoda ta adnotacja
@scheduled(fixedDelay = 3600000)
public int keyGenerator()

pszczepcio replied:
W zadaniu jest : "wygenerowanie losowego klucza ważnego godzinę po podaniu poprawnych danych użytkownika "
czyli użytkownik może sobie ten klucz sam generować. np. poprzez zapytanie??

klikuskodilla commented on src/main/java/com/kodilla/ecommercee/domain/user/User.java:
```diff
@@ -0,0 +1,74 @@
+        userKey = keyGenerator();
+    }
+
+    public int keyGenerator(){
```
czy

>  keyValue=random.nextInt(89999)+10001;

 nie będzie generował takiego samego zakresu wartości?

pszczepcio committed:
Merge branch 'master' of https://github.com/klikuskodilla/project_jdp_1904_01

pszczepcio committed:
Merge branch 'master' into JDP190401-15

pszczepcio committed:
- improving the generator function
- adding a field specifying the date and time of key generation
- there will be a method for generating a new key and its creation time

Makedo81 commented on src/main/java/com/kodilla/ecommercee/domain/user/User.java:
```diff
@@ -54,21 +54,16 @@ public User(String userName){
+        timeGenerateKey = new Date();
     }
 
     public int keyGenerator(){
```
zeby klucz byl generowany co godzine to powinno byc nad metoda ta adnotacja
@Scheduled(fixedDelay = 3600000)
public int keyGenerator()

klikuskodilla replied:
W wymaganiu jest napisane że klucz ma być ważny przez godzine, a nie generowany co godzinę. Metoda generowania klucza to poprawne zalogowanie się

pszczepcio replied:
To w klasie user dodać dodatkowo pole login i password? i jak uzytkownik przesle w zapytaniu prawidlowe dane to wtedy jest wygenerowany klucz?

Makedo81 replied:
a jezeli po godzinie klucz wygasnie to co ma sie dziac?

klikuskodilla replied:
Tak, można tak zrobić. login to będzie userName, wiec nie trzeba dodawać, tylko samo password.

Makedo81 commented on src/main/java/com/kodilla/ecommercee/domain/user/User.java:
```diff
@@ -54,21 +54,16 @@ public User(String userName){
     }
 
     public int keyGenerator(){
-        int keyValue = 0;
```
klucz powinien byc generowany po podaniu poprawnych danych uzytkownika czyli  trzeba chyba dopisac warunek if (status ==true) {  tutaj funkcja generujaca key  }

Makedo81 commented on src/main/java/com/kodilla/ecommercee/domain/user/User.java:
```diff
@@ -54,21 +54,16 @@ public User(String userName){
+    }
 
-        for (int i = 0 ; i < keyNumbers.length ; i++) {
-            if (keyNumbers[4] == 0){
```
Ten warunek nie potrzebny

Makedo81 reviewed (commented):
keyValue bedzie zwracane przez metode generateKey() i zapisywane w bazie danych i stamtad bedzie mozna go pobrac w zapytaniu. A zapytanie bedzie musialo byc warunkowe i bedzie chyba zalezne os wartosci STATUS, a wartosc STATUS okreslana na podstawie login i haslo

pszczepcio commented:
Ten warunek dodam do metody updateKeyNumber() . Tą metodę będzie się używało do generowania nowego klucza. A sama metoda  keyGenerator() jest używana tylko przy utworzeniu użytkownika.

pszczepcio committed:
- added a conditions to method updateKeyNumber

Makedo81 commented on src/main/java/com/kodilla/ecommercee/domain/user/User.java:
```diff
@@ -63,7 +63,11 @@ public int keyGenerator(){
     }
 
     public void updateKeyNumber(){
```
Wydaje mi sie ,ze ta metoda w ogole nie jest potrzebna. Wystarczy tylko jedna metoda generujaca klucz przypisana do  userKey. Zwraca wartosc userKey i zapisuje go w bazie danych. Tylko nie wiem czy  ta wartosc userKey ma byc nadpisywana w bazie danych w momencie gdy tworzymy nowy userKey, czy dodajemy nowy userKey z nowym id.

Makedo81 commented on src/main/java/com/kodilla/ecommercee/domain/user/User.java:
```diff
@@ -54,21 +54,16 @@ public User(String userName){
         this.userName = userName;
```
Moze przydaloby sie tutaj dodac tez pole login i haslo, bo to tez sa atrybuty nowego uzytkownika i wtedy dopisac metode boolean generujaca status true albo false

klikuskodilla commented on src/main/java/com/kodilla/ecommercee/domain/user/User.java:
```diff
@@ -0,0 +1,73 @@
+    }
+
+    public void updateKeyNumber(){
+        if (status == true) {
```
updateKeyNumber nie powinna decydować o tym czy może zmieniać klucz. Taka logika powinna być przeniosiona w inne miejsce np UserService(którego jeszcze nie ma wiec nie pisałbym logiki odpowiedzialnej za logowanie) która decyduje o tym czy logowanie powiodło

pszczepcio replied:
ok. To dodaje pole password do klasy a logiką zajmę się później.

pszczepcio committed:
Merge branch 'master' of https://github.com/klikuskodilla/project_jdp_1904_01

pszczepcio committed:
- added field "PASSWORD"
- delete method updateKeyNumber()

pszczepcio committed:
Merge branch 'master' into JDP190401-15

pszczepcio committed:
- added field "PASSWORD"
- delete method updateKeyNumber()
